tests/data_generation/helper/induce_causal_effect.py

```
@@ -1,555 +1,303 @@
+            trigger_effects = []
+            for i, trigger_code in enumerate(outcome_cfg.trigger_codes):
+                is_present = trigger_code in history_codes
+                weight = outcome_cfg.trigger_weights[i]
+                trigger_effects.append(weight if is_present else 0.0)
+
+            # Factual scenario: was the patient actually exposed at assessment time?
+            factual_exposure_present = self.config.exposure.code in history_codes
+
+            # Helper to compute probability based on exposure status
+            def _compute_prob(is_exposed: bool) -> float:
+                logit_p = logit(outcome_cfg.p_base)
```

Ruff (0.11.9)

Function definition does not bind loop variable `outcome_cfg`

(B023)